### DISCUSSION TITLE
POC for functional-like API

### DISCUSSION THREAD

**Discussion by cevr** [statelyai/xstate#1451] [General]
Hey there!

Here's a POC for a functional-like API for xstate.

What do you think? 

https://gist.github.com/cevr/8f984684fd24be94ec5c4d1c8a6f263c

**Comment by davidkpiano** [comment]
Neat! I was playing around with similar Swift-UI-inspired syntax ("XState builder") here: https://gist.github.com/davidkpiano/ae330406e630c63fe22022cdceb36f32 

The difference with this is that it's a bit more flexible in how you define the machine/state structure.

**Reply by cevr** [reply]
Ah cool! I like that swift-UI syntax a lot haha
